geowebcache/s3storage/src/main/java/org/geowebcache/s3/S3BlobStoreInfo.java

```diff
@@ -334,7 +337,38 @@ public BlobStore createInstance(TileLayerDispatcher layers, LockProvider lockPro
         checkState(
                 isEnabled(),
                 "Can't call S3BlobStoreConfig.createInstance() is blob store is not enabled");
-        return new S3BlobStore(this, layers, lockProvider);
+        final GeoWebCacheEnvironment gwcEnvironment =
+                GeoWebCacheExtensions.bean(GeoWebCacheEnvironment.class);
+        return new S3BlobStore(this.clone(gwcEnvironment, true), layers, lockProvider);
+    }
+
+    public S3BlobStoreInfo clone(
+            GeoWebCacheEnvironment gwcEnvironment, Boolean allowEnvParametrization) {
+        S3BlobStoreInfo blobStore = SerializationUtils.clone(this);
+
+        if (allowEnvParametrization && gwcEnvironment != null) {
+            blobStore.setName((String) gwcEnvironment.resolveValue(getName()));
+            blobStore.setEnabled((Boolean) gwcEnvironment.resolveValue(isEnabled()));
```

Comment:
Hum... are you sure the environment will provide back Integer, Access or Integer? I don't see how it can determine the desired type. Most of the env parametrization in GWC needs strings, so the other examples are just fine casting to String (which is, I believe, what you'll actually get back).

Can you please modify the tests in order to test parametrization of all types of properties, so that we can be sure it's actually working?

Also, in order to host parametric values, you'll need to change all the properties in the INfo to be String, I don't see how you can make them parametric otherwise.